build_tools/github_actions/configure_ci.py

```diff
@@ -17,6 +17,7 @@
   * WINDOWS_TEST_LABELS (optional): Comma-separated list of test labels to test
   * WINDOWS_USE_PREBUILT_ARTIFACTS (optional): If enabled, CI will only run Windows tests
   * BRANCH_NAME (optional): The branch name
+  * BUILD_VARIANT (optional): The build variant to run (ex: release, asan)
```

Comment:
Can this be a list of variants instead of a top level toggle? I'd like to keep this script (and whatever triggers it) flexible:

* Inputs: files changed, job triggers, optional variables for opt in/out
* Outputs: which jobs run

Forking at the top level may warrant a separate `configure_ci_asan.py` script (and we could refactor so code is shared between each)